pkg/controllers/machine_controller.go

```diff
@@ -765,7 +765,7 @@ func (r *MachineReconciler) setDomainResources(machine *api.Machine, domain *lib
 		}
 	}
 
-	cpu := uint(machine.Spec.CpuMillis / 1000)
+	cpu := uint(machine.Spec.Resources.CPU().Value() / 1000)
```

Comment:
this logic is wrong. 8.6 will rounded down to 8 what doesn't make sense for me. I prefer rounded up.

@afritzler do we need support millis unit? Cannot we support just whole cores? 1, 2, 3, 4, 5, 6, 7, 8, ...